python/src/trezorlib/firmware/core.py

```diff
@@ -129,19 +130,26 @@ def code_hashes(self) -> t.List[bytes]:
         Assume that the first `code_offset` bytes of `code` are taken up by the header.
         """
         hashes = []
+
+        if self.HASH_PARAMS.chunk_size is None:
+            internal_name = Model.from_hw_model(self.header.hw_model).value.decode()
+            trezor_model = by_internal_name(internal_name)
+            if trezor_model is None:
+                return hashes
+            chunk_size = trezor_model.chunk_size
+        else:
+            chunk_size = self.HASH_PARAMS.chunk_size
```

Comment:
...and then this can just be
```
params = Model.from_hw_model(self.header.hw_model).hash_params()
```

Reply:
i still need to distinguish legacy and core, because (at least in tests) both have `hw_model` field zeroed and thus would default to T in `from_hw_model`. Added a `get_hash_params` function for that purpose. It is overridden in `LegacyV2Firmware`
https://github.com/trezor/trezor-firmware/commit/9544f53afd8ba2e38b7b8c7ede6acee73cbd0d74